test/PandoraInterface.cxx

```diff
@@ -1038,12 +1039,17 @@ void CreateSPMCParticles(const LArSPMC &larspmc, const pandora::Pandora *const p
         const std::string reaction = GetNuanceReaction((*larspmc.m_ccnc)[nuIndex], (*larspmc.m_mode)[nuIndex]);
         mcParticleParameters.m_nuanceCode = GetNuanceCode(reaction);
 
-        // Set unique parent integer address using trackID
-        const int trackID = (*larspmc.m_mcp_id)[i];
+        // Set unique parent integer address using trackID. Need to add a large enough
+        // offset of 10^6 to make these unique when we have more than 1 neutrino per event.
+        // The mcp_id's reset (to zero) per neutrino interaction vertex, and the offset
+        // should allow up to 10^6 hits for each neutrino. The true neutrino IDs (nuID)
+        // found earlier are offset by 10^8, which should allow unique trackID's for up to
+        // 100 neutrino interactions per event, each containing up to 10^6 hits
+        const int offsetID = 1000000 * nuIndex;
```

Comment:
Is it worth defining a separate track id offset constant, in a similar fashion to the nu id offset, in case it ever needs to be reused anywhere?

Reply:
A check is now made to make sure that the hit IDs are within the various limits. A maximum offset factor of 10^6 is specified before we loop over the hits, but we need to recalculate the offset for each hit since this depends on what the neutrino number is (0 to N-1 for N neutrinos in the event).